UIROLES-73: assigned users are not checked when open Pluggable find-user component

aidynoJ opened the pull request:
Refs [UIROLES-73](https://folio-org.atlassian.net/browse/UIROLES-73).
 Instead of using variable `users` which is array of object that includes `id`, the variable `usersData` was provided as [] { fullName, patronGroup }.
This PR also fixes a small issue introduced by UIROLES-46. The basename={baseUrl} provided to the Router component prevents users from navigating out of the basename, which is necessary for AccordionUsers using TextLink

```
<TextLink to={`/users/preview/${i.id}`}>
        {getFullName(i)}
      </TextLink>
```

zburke reviewed (commented):
I'd really like to get away from string parsing to handle routes. `router.path.split` is not a pattern I see in other applications. Let's try to avoid it here, too. 

It isn't clear to me how the changes in this PR resolve the bug, nor exactly what the bug was. "Fix checkboxes" doesn't provide enough detail: were they ticked incorrectly, were they unticked incorrectly, were displayed correctly but handled incorrectly?

aidynoJ commented on src/settings/components/CreateEditRole/EditRole.js:
```diff
@@ -18,6 +18,11 @@ const EditRole = () => {
+    const basePath = router.path.split(':')[0];
+    return basePath.endsWith('/') ? `${basePath}${id}` : `${basePath}/${id}`;
+  };
+
```
Fixed, I've removed all constructions like that. Now, the props.match.path is directed to the root src/index.js component, and there's no longer a hard-coded basePath for defining router components.

aidynoJ commented on src/settings/components/RoleDetails/AccordionUsers.js:
```diff
@@ -53,11 +53,11 @@ const AccordionUsers = ({ roleId }) => {
       }
       displayWhenOpen={<AssignUsers
-        selectedUsers={usersData}
+        selectedUsers={users}
```
The issue with this ticket is that the wrong variable was accidentally provided. Instead of using `users` which is array of object that includes `id`, the variable usersData was provided as [] { fullName, patronGroup }.

aidynoJ commented:
> I'd really like to get away from string parsing to handle routes. `router.path.split` is not a pattern I see in other applications. Let's try to avoid it here, too.
> 
> It isn't clear to me how the changes in this PR resolve the bug, nor exactly what the bug was. "Fix checkboxes" doesn't provide enough detail: were they ticked incorrectly, were they unticked incorrectly, were displayed correctly but handled incorrectly?

The fix of the main issue of this ticket was trivial, just providing correct variable to AssignUsers.  
I've changed description, and I have removed all constructions like route.path.split , etc. 
Thank you for reviewing

zburke commented on src/settings/components/SettingsPage/SettingsPage.js:
```diff
@@ -29,7 +29,7 @@ const SettingsPage = () => {
     <PaneMenu>
       <Button
-        to="/create"
+        to={path + '/create'}
```
This is fine, but the Stripes convention is interpolation > concatenation, so:
```
{`${path}/create`}
```
